src/relation_cache.c

```diff
@@ -177,10 +193,19 @@ update_relation_cache(Oid relid)
 	update_relation_entry(relid, &relation_entry_data, &relid_entry_data);
 
 	LWLockAcquire(diskquota_locks.relation_cache_lock, LW_EXCLUSIVE);
-	relation_entry = hash_search(relation_cache, &relation_entry_data.relid, HASH_ENTER, NULL);
+	relation_entry = RELATION_CACHE_ENTER(&relation_entry_data.relid, NULL);
+	if (relation_entry == NULL)
+	{
+		LWLockRelease(diskquota_locks.relation_cache_lock);
+		return;
+	}
 	memcpy(relation_entry, &relation_entry_data, sizeof(DiskQuotaRelationCacheEntry));
-
-	relid_entry = hash_search(relid_cache, &relid_entry_data.relfilenode, HASH_ENTER, NULL);
+	relid_entry = RELID_CACHE_ENTER(&relid_entry_data.relfilenode, NULL);
+	if (relid_entry == NULL)
```

Comment:
```suggestion
	if (relid_entry != NULL) *relid_entry = relid_entry_data;
```

Reply:
Are you sure we can do this? Because if there is no space in relation_cache, then there is no space in active_tables_map and diskquota won't work correctly anymore.

Reply:
why not? it is optimized (by compiler) version of memcpy only

Reply:
we may also do so
```suggestion
	if (relid_entry != NULL) memcpy(relid_entry, &relid_entry_data, sizeof(DiskQuotaRelidCacheEntry));
```

Reply:
I mean, this is not just a check before copying if there is no space, we exit the function. how much do we want to continue executing the function if there is no place for the hash map? and how correct would that be?

Reply:
You may add something like
```c
	if (relation_entry == NULL) return;
	prelid = get_primary_table_oid(relid, false);
```

Reply:
I don't see the point in searching in the second map if we didn't find it in the first one. since we add entries in pairs, if we didn't find them in the first one, we probably won't find them in the second one either.

Reply:
May be put them into one map?